app/Features/QueryRandomAnimeHandler.php

```diff
@@ -5,28 +5,31 @@
 use App\Anime;
 use App\Contracts\RequestHandler;
 use App\Dto\QueryRandomAnimeCommand;
+use App\Http\Resources\V4\AnimeCollection;
 use App\Http\Resources\V4\AnimeResource;
 use Spatie\LaravelData\Optional;
 
 /**
- * @implements RequestHandler<QueryRandomAnimeCommand, AnimeResource>
+ * @implements RequestHandler<QueryRandomAnimeCommand, AnimeResource|AnimeCollection>
```

Comment:
Just an idea: Wouldn't this be a bit confusing for people? This would change the so far expected response. With this change we are expecting users to know that if they add `limit=15` for example, they would get a list. Wouldn't it be more "backwards compatible" to create a new endpoint which can return a list? The other option is to make this endpoint always return a list. There should be no varying response format based on query string parameters.

Reply:
@pushrbx Agreed. We can opt for a separate endpoint, unsure of what to call it though. Any suggestions?

Reply:
`/random/list/anime` maybe?